src/database/services/severa-api-service.ts

```diff
@@ -39,7 +45,86 @@ export const CreateSeveraApiService = (): SeveraApiService => {
           `Failed to fetch flextime: ${response.status} - ${response.statusText}`,
         );
       }
+      return response.json();
+    },
+
+    /** 
+     * Get resource allocation by userId
+     */
+       getResourceAllocation: async (severaUserId: string) => {
+        const url: string = `${baseUrl}/v1/users/${severaUserId}/resourceallocations/allocations`;
+        const response = await fetch(url,{
+            method: "GET",
+            headers: {
+                Authorization: `Bearer ${await getSeveraAccessToken()}`,
+                "Client_Id": process.env.SEVERA_DEMO_CLIENT_ID,
+                "Content-Type": "application/json",
+            },
+        });
+        if (!response.ok) {
+            throw new Error(
+                `Failed to fetch resource allocation: ${response.status} - ${response.statusText}`,
+            );
+        }
+        return response.json();
+    },
+
+    /**
+     * Gets phases by userId
+     */
+    getPhasesBySeveraProjectId: async (severaProjectId: string) => {
+      const url: string = `${baseUrl}/v1/projects/${severaProjectId}/phaseswithhierarchy`;
+
+      const response = await fetch(url, {
+        method: "GET",
+        headers: {
+          Authorization: `Bearer ${await getSeveraAccessToken()}`,
+          "Client_Id": process.env.SEVERA_DEMO_CLIENT_ID,
+          "Content-Type": "application/json",
+        },
+      });
+      if (!response.ok) {
+        throw new Error(
+          `Failed to fetch phases: ${response.status} - ${response.statusText}`,
+        );
+      }
+      return response.json();
+    },
+
+    /**
+     * Gets work hours by userId
+     */
+    getWorkHoursBySeveraId: async (severaProjectId?:string, severaUserId?: string, severaPhaseId?: string, startDate?: string, endDate?: string) => {
+      let url: string;
 
+      if (severaProjectId) {
+        url = `${baseUrl}/v1/projects/${severaProjectId}/workhours`;
+      } else if (severaUserId) {
+        url = `${baseUrl}/v1/users/${severaUserId}/workhours`;
+      } else {
+        url = `${baseUrl}/v1/workhours`;
+      }
```

Comment:
If user passes 1st params such as projectID, it will use the url workHours for projects. Then, within getWorkHoursHandler, if user passes 2nd params such as userID, getWorkHoursHandler than will fetch the workHours from that user in that project. if user passes 3rd params such as phaseID, getWorkHoursHandler will do the same as before to get that data. However, it's exactly what you said that it is filtering twice which is not good. Another filtering method is in developing and will be discussed with you later on